src/viam/sdk/common/proto_value.hpp

```diff
@@ -328,17 +328,17 @@ void to_proto(const ProtoList& vec, google::protobuf::Value* v);
 void to_proto(const ProtoStruct& m, google::protobuf::Value* v);
 void to_proto(const ProtoValue& t, google::protobuf::Value* v);
 
-ProtoStruct struct_to_map(google::protobuf::Struct const* s);
+void struct_to_map(google::protobuf::Struct const* s, ProtoStruct& m);
```

Comment:
What motivated changing this API?

Reply:
The old way was erroring out on compilation with an infinite loop of template insantiations since we had both

```
ProtoStruct struct_to_map(protobuf::Struct const* s);

template<typename Struct = protobuf::Struct>
ProtoStruct struct_to_map(const Struct& s);
```
absent any constraints, such as `enable_if_t<!std::is_pointer<std::remove_reference_t<T>>{}>`, I believe they ended up calling each other over and over.

This fixed that ambiguity without `enable_if`, and it sort of matches the implied semantic convention that the overloads taking pointer are not really for normal use